agent/engine/task_manager.go

```diff
@@ -151,6 +151,7 @@ func (mtask *managedTask) overseeTask() {
 		llog.Debug("Marking done for this sequence", "seqnum", mtask.StopSequenceNumber)
 		mtask.engine.taskStopGroup.Done(mtask.StopSequenceNumber)
 	}
+	go mtask.releaseIPInIPAM()
```

Comment:
Is this idempotent?  You may end up with multiple calls/partial completion if there's an agent restart in the middle.

Reply:
This will ensure the ip is removed if agent restarted, but could emit an error in the ipam logs if  this was called twice before and after restart. One possible solution would be remove the eniattachment from state here and check the eniattachment before calling releasing. But there is also a race condition where restart happened in the middle of saving state. Any better suggestion, otherwise I will just add a TODO in the code and we can come back to that later.

Reply:
If the only downside is an error in the IPAM logs, I'm okay with adding a TODO.